### ISSUE TITLE
OSC won't work a few minutes after playing and never work again before restarting Unreal Editor.

### ISSUE THREAD

**Issue by knagata** [monsieurgustav/UE4-OSC#19]
Hi,

I am facing a problem that OSC won't work a few minutes after playing. Actually the problem is very similar to this issue (https://github.com/monsieurgustav/UE4-OSC/issues/17).
But the weird point is that I am running the project on UE4.10.2. The conclusion in that issue was the version.

I am not sure but, I checked Output Log and find a message that related to this problem.
>LogStats:Warning: MetaData mismatch. Did you assign a stat to two groups? New //STATGROUP_Threads//OSCListener///Thread_f33b_0///####STATCAT_Advanced#### old //STATGROUP_Threads//OSCListener///Thread_1c5b7_0///####STATCAT_Advanced####

I have the problem under these situation:
* OSX 10.10.5
* Xcode 7.2
* UE4.10.2
* Latest UE4-OSC

I just in case put an image of my blueprint.
<img width="1920" alt="screen shot 2016-01-18 at 05 39 14" src="https://cloud.githubusercontent.com/assets/917520/12379839/f6b2ac64-bda5-11e5-944c-6bc8174f6777.png">

If you have noticed anything wrong, please give me some information. And also, so I am actually really the beginner of UE, if there is lack of information to identify the cause, please let me know.

Thanks in advance.

Knag

**Comment by monsieurgustav** [comment]
Hi !
This commit
https://github.com/monsieurgustav/UE4-OSC/commit/65ee771d80e1bf2c58b88189de1c72683cca5fd1
includes a fix. Do you have it ?

Don't you have any "LogOSC" message ?
Try to set the verbosity to "VeryVerbose" to have more information.

**Comment by knagata** [comment]
Thank you so much for the quick reply.

I changed the verbosity and get the following message when the program is working:
>LogOSC:Verbose: Received: L3RyaWdnZXIAAAAALGYAAEDgGGw=
>LogBlueprintUserMessages: [osc_34] /trigger
>LogOSC:Verbose: Received: L3RyaWdnZXIAAAAALGYAAEC0Dw4=
>LogBlueprintUserMessages: [osc_34] /trigger

And then, after stopping working, I got these messages:
>LogOSC:Verbose: Received: L3RyaWdnZXIAAAAALGYAAD+wyYo=
>LogOSC:Verbose: Received: L3RyaWdnZXIAAAAALGYAAD9xQWw=
>LogOSC:Verbose: Received: L3RyaWdnZXIAAAAALGYAAD8b5kg=

The messages seem to be received, but "Event On Osc Received" component stopped working.

I am sure that the plugin is the latest version which is released on Dec.31th (or Jan.1st).
So the commit is certainly included.

Just in case, I will try to download the plugin with the commit and build again to test.
But basically the plugin is up to date.

**Comment by monsieurgustav** [comment]
OK.
Could you add 2 lines in the plugin code and send the log when it goes
wrong ?

In OscDispatcher.cpp, line 187 (in the if scope) :
UE_LOG(LogOSC, Verbose, TEXT("Task sent");

In OscDispatcher.cpp, line 206 :
UE_LOG(LogOSC, Verbose, TEXT("Task done");

**Comment by knagata** [comment]
I've changed code like this:
<img width="939" alt="screen shot 2016-01-19 at 02 39 40" src="https://cloud.githubusercontent.com/assets/917520/12398318/3480ee4c-be56-11e5-9aa9-15fffffe3182.png">

And got this error when compiling:
<img width="803" alt="screen shot 2016-01-19 at 02 40 47" src="https://cloud.githubusercontent.com/assets/917520/12398339/6ab6c0ea-be56-11e5-8469-6a4f218b6f3b.png">

Thank you so much for your help and sorry for bothering you.

**Comment by monsieurgustav** [comment]
My mistake:

In OscDispatcher.cpp, line 187 (in the if scope) :
UE_LOG(LogOSC, Verbose, TEXT("Task sent"));

In OscDispatcher.cpp, line 206 :
UE_LOG(LogOSC, Verbose, TEXT("Task done"));

**Comment by knagata** [comment]
Hi,

I realized the mistake, it was jumping to the conclusion...
Sorry for my impatience.

I inserted these commands in the code and tested how the code works.
It's really weird but, when I tested how the project works, I never have the problem again...

I can understand that these 2 commands do nothing but giving me LogOSCs in OutputLog.
For more test, I will leave the project running whole day to check whether the problem occurs again.
